website/www/site/content/en/documentation/io/built-in/snowflake.md

```diff
@@ -362,3 +837,209 @@ static SnowflakeIO.CsvMapper<GenericRecord> getCsvMapper() {
+
+### Set up 
+Please see [Apache Beam with Flink runner](https://beam.apache.org/documentation/runners/flink/) for a setup. The specific setup for current version of snowflake is following:
+
+1. Setup a Flink cluster by following the Flink [Setup Quickstart](https://ci.apache.org/projects/flink/flink-docs-release-1.10/getting-started/tutorials/local_setup.html) or [Setting up Apache Flink on Mac OS X](https://streambench.wordpress.com/2017/10/26/setting-up-apache-flink-on-mac-os-x/)
+2. Download Job server image:
+`docker pull gcr.io/snowflake-poli/apachebeam_flink1.10_job_server:snowflake`
+3. Download Apache Beam Java SDK image:
+`docker pull gcr.io/snowflake-poli/apachebeam_java_sdk:2.20.0.dev`
+4. Change tag of downloaded Java SDK image to make the whole setup work:
+{{< highlight >}}
+docker tag gcr.io/snowflake-poli/apachebeam_java_sdk:2.20.0.dev apache/beam_java_sdk:2.20.0.dev
+{{< /highlight >}}
+5. Start Job server:
+{{< highlight >}}
+docker run -p 8099:8099 -p 8098:8098 -p 8097:8097 gcr.io/snowflake-poli/apachebeam_flink1.10_job_server:snowflake
+{{< /highlight >}}
+6. Download [Apache Beam Python SDK](https://storage.cloud.google.com/snowflake_artifacts/apache_beam-2.21.0-py2-none-any.whl?_ga=2.24638847.-471657054.1583857613).
+7. Install python Apache Beam Python SDK using python 2.7:
+`python -m pip install apache_beam-2.21.0-py2-none-any.whl`
+8. Run your job:
+`python script.py`
+### Reading from Snowflake 
+One of the functions of SnowflakeIO is reading Snowflake tables - either full tables via table name or custom data via query. Output of the read transform is a [PCollection](https://beam.apache.org/releases/pydoc/2.20.0/apache_beam.pvalue.html#apache_beam.pvalue.PCollection) of user-defined data type.
+#### General usage
+{{< highlight >}}
+OPTIONS = [
+   "--runner=FlinkRunner",
+   "--flink_version=1.10",
+   "--flink_master=localhost:8081",
+   "--environment_type=LOOPBACK"
+]
+
+with TestPipeline(options=PipelineOptions(OPTIONS)) as p:
+   (p
+       | ReadFromSnowflake(
+           server_name=<SNOWFLAKE SERVER NAME>,
+           username=<SNOWFLAKE USERNAME>,
+           password=<SNOWFLAKE PASSWORD>,
+           o_auth_token=<OAUTH TOKEN>,
+           private_key_path=<PATH TO P8 FILE>,
+           raw_private_key=<PRIVATE_KEY>
+           private_key_passphrase=<PASSWORD FOR KEY>,
+           schema=<SNOWFLAKE SCHEMA>,
+           database=<SNOWFLAKE DATABASE>,
+           staging_bucket_name=<GCS BUCKET NAME>,
+           storage_integration_name=<SNOWFLAKE STORAGE INTEGRATION NAME>,
+           csv_mapper=<CSV MAPPER FUNCTION>,
+           table=<SNOWFALKE TABLE>,
+           query=<<IF NOT TABLE THEN QUERY>>,
+           expansion_service=<EXPANSION SERVICE ADDRESS>))
```

Comment:
I know that @piotr-szuberski  added support for other parameters like role etc. We should add it here.

Reply:
TODO

Reply:
fixed